internal/resource_remote_file.go

```diff
@@ -0,0 +1,703 @@
+		if errors.Is(err, client.ErrTaskNotFound) {
+			return nil
+		}
+
+		errs = append(errs, fmt.Errorf("stop download task: %w", err))
+	}
+
+	if err := c.EraseDownloadTask(ctx, taskID); err != nil {
+		if errors.Is(err, client.ErrTaskNotFound) {
+			return nil
+		}
+
+		errs = append(errs, fmt.Errorf("erase download task: %w", err))
+	}
+
+	return errors.Join(errs...)
+}
+
+func deleteFilesIfExist(ctx context.Context, c client.Client, paths ...string) (error) {
+	filesToDelete := make([]string, 0, len(paths))
+
+	var errs []error
+	for _, path := range paths {
+		_, err := c.GetFileInfo(ctx, path)
+		if err != nil {
+			if !errors.Is(err, client.ErrPathNotFound) {
+				errs = append(errs, err)
+			}
+
+			continue
+		}
+
+		filesToDelete = append(filesToDelete, path)
+	}
+
+	if len(errs) > 0 {
+		return errors.Join(errs...)
+	}
+
+	if len(filesToDelete) == 0 {
+		return nil
+	}
+
+	task, err := c.RemoveFiles(ctx, filesToDelete)
+	if err != nil {
+		return fmt.Errorf("remove files: %w", err)
+	}
+
+	if err := waitForFileSystemTask(ctx, c, task.ID); err != nil {
+		var taskError *taskError
+		if errors.As(err, &taskError) && taskError.errorCode == string(freeboxTypes.FileTaskErrorFileNotFound) {
+			return nil
+		}
+
+		return fmt.Errorf("wait for file system task: %w", err)
+	}
+
+	return nil
+}
+
+// verifyFileChecksum verifies the checksum of the file.
+func verifyFileChecksum(ctx context.Context, client client.Client, path string, checksum string) (diagnostics diag.Diagnostics) {
+	hMethod, hValue := hashSpec(checksum)
+
+	hash, err := fileChecksum(ctx, client, path, hMethod)
+	if err != nil {
+		diagnostics.AddError("Failed to read hash", err.Error())
+		return
+	}
+
+	if hash != hValue {
+		diagnostics.AddError("Checksum mismatch", fmt.Sprintf("Expected %q, got %q", checksum, hash))
+	}
+
+	return
+}
+
+// fileChecksum verifies the checksum of the file.
+func fileChecksum(ctx context.Context, client client.Client, path string, hashType string) (string, error) {
+	if hashType == "" {
+		hashType = string(freeboxTypes.HashTypeSHA256)
+	}
+
+	task, err := client.AddHashFileTask(ctx, freeboxTypes.HashPayload{
+		HashType: freeboxTypes.HashType(hashType),
+		Path:     freeboxTypes.Base64Path(path),
+	})
+
+	if err != nil {
+		return "", fmt.Errorf("hash file task: %w", err)
+	}
+
+	if err := waitForFileSystemTask(ctx, client, task.ID); err != nil {
+		return "", fmt.Errorf("wait for file system task: %w", err)
+	}
+
+	hash, err := client.GetHashResult(ctx, task.ID)
+	if err != nil {
+		return "", fmt.Errorf("get hash result: %w", err)
+	}
+
+	return hash, nil
+}
+
+type taskError struct {
+	taskID int64
+	errorCode string
+}
+
+func (e *taskError) Error() string {
+	return fmt.Sprintf("task %d failed with error code %q", e.taskID, e.errorCode)
+}
+
+// waitForFileSystemTask waits for the system task to complete.
+func waitForFileSystemTask(ctx context.Context, client client.Client, taskID int64) (error) {
+	tick := time.NewTicker(5 * time.Second)
```

Comment:
That delay in between polls should probably be configurable at the resource level or something ; could help speed up the tests

Same goes for `waitForDownloadTask`.